lib/Xud.ts

```diff
@@ -112,32 +89,28 @@ class Xud extends EventEmitter {
       this.logger.info(`Local nodePubKey is ${this.nodeKey.nodePubKey}`);
 
       // initialize pool and start listening/connecting only once other components are initialized
-      const lndPubKeys: { [currency: string]: string | undefined } = {};
-      for (const currency in this.lndClients) {
-        lndPubKeys[currency] = this.lndClients[currency]!.pubKey;
-      }
-
       await this.pool.init({
         version,
-        lndPubKeys,
+        lndPubKeys: this.swapClientManager.getLndPubKeys(),
         pairs: this.orderBook.pairIds,
         nodePubKey: this.nodeKey.nodePubKey,
-        raidenAddress: this.raidenClient.address,
+        raidenAddress: this.swapClientManager.getRaidenAddress(),
       }, this.nodeKey);
 
       this.service = new Service({
         version,
         orderBook: this.orderBook,
-        lndClients: this.lndClients,
-        raidenClient: this.raidenClient,
+        swapClientManager: this.swapClientManager,
         pool: this.pool,
         swaps: this.swaps,
         shutdown: this.beginShutdown,
       });
 
-      if (!this.raidenClient.isDisabled()) {
+      if (!this.config.raiden.disable) {
```

Comment:
We no longer have `this.raidenClient` within the context of `Xud.ts`. I think it makes sense to enable the `httpServer` based on the configuration value of `raiden.disable` because if we're relying on the dynamic `raidenClient.isDisabled()` it could be in a state of trying to reconnect. However, once the reconnection happens we don't have logic to start the `httpServer`.

Reply:
If it's trying to reconnect, it should have the `Disconnected` status instead of `Disabled`. `Disabled` means it's off and staying off, and although currently the only way raiden is disabled is via config, that could change one day (the lnd client can be disabled if we can't load the tls/macaroon files).

Could we expose the raiden status via the swap client manager? We might need it for other purposes down the road as well.

Reply:
As discussed in private I ended up exposing the whole `raidenClient` instance and removing `SwapClientManager.getRaidenInfo/getRaidenAddress` as they were redundant.